myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+                    ]
+                    old_fat = old_goals["items"][0]["default_goal"]["fat"]
+                    old_protein = old_goals["items"][0]["default_goal"]["protein"]
+
+                    # If old and new values are in diffrent units then convert old value to new unit
+                    if not old_energy_unit == energy_unit:
+                        if old_energy_unit == "kilojoules" and energy_unit == "calories":
+                            old_energy_value *= 0.2388
+                            old_energy_unit = "calories"
+                        elif old_energy_unit == "calories" and energy_unit == "kilojoules":
+                            """FROM MFP JS
+                            if (energyUnit === 'kilojoules') {
+                                calories *= 4.184;
+                            }
+                            """
+                            old_energy_value *= 4.184
+                            old_energy_unit = "kilojoules"
+                        else:
+                            raise ValueError
+
+                    carbohydrates = energy * old_carbohydrates / old_energy_value
+                    protein = energy * old_protein / old_energy_value
+                    fat = energy * old_fat / old_energy_value
+            # If percentage values were provided check
+            else:
+                if energy is None:
+                    logger.warning("No energy value and no macro values provided! - Not able to update goals without a mandotory amount of information.")
+                    return False
+                else:
+                    if percent_carbohydrates + percent_protein + percent_fat == 100.0:
+                        carbohydrates = energy * percent_carbohydrates / 100.0 / 4
+                        protein = energy * percent_protein / 100.0 / 4
+                        fat = energy * percent_fat / 100.0 / 9
+                        if energy_unit == "kilojoules":
+                            carbohydrates = round(carbohydrates / 4.184, 2)
+                            protein = round(protein / 4.184, 2)
+                            fat = round(fat / 4.184, 2)
+                    else:
+                        logger.warning("Provided percentage values are not plausible - Not able to update goals.")
+                        return False
+        else:
+            macro_energy = carbohydrates * 4 + protein * 4 + fat * 9
+            if energy_unit == "kilojoules":
+                macro_energy *= 4.184
+            # Compare energy values and set it correctly due to macros. Will also fix if no energy_value was provided.
+            if energy != macro_energy:
+                logger.warning(
+                    "Provided energy value and calculated energy value from macros do not match! Will override!"
+                )
+                energy = macro_energy
+
+        # Build payload based on observed browser behaviour
+        # TODO Insert additional micro nurtitions
```

Comment:
Could I convince you into removing these TODOs?  I understand that you're trying to guide future writers, but very often comments like this end up just becoming noise.

Reply:
Done.